### ISSUE TITLE
Allow configuration of default JS file types

### ISSUE THREAD

**Issue by justin808** [eslint/eslint#2419]
Per https://github.com/eslint/eslint/issues/2388#issuecomment-98183996:

Let's allow configuration of the file suffixes, maybe in the .eslintrc file.

The problem was that I was not including the jsx files at the command line. Almost all my JS files have suffix JSX.

So this works:
```
#!/bin/sh
eslint . --ext .jsx,.js
jscs .
```

On a separate note, if team members don't want to install eslint globally, any disadvantage to doing this:

```
npm run eslint . -- --ext .jsx,.js
```

**Comment by nzakas** [comment]
I'm not sure we can actually add extensions into the config file due to the order of operations in ESLint. I do wonder, though, if we should just add `.jsx` as a default in addition to `.js`?

**Comment by justin808** [comment]
I would say that's it's safe to by default include both.

Another alternative would be to allow a file ".eslintinclude" that is analogous to the ignore file.

**Comment by michaelficarra** [comment]
@nzakas I don't know if that makes sense. I could have eslint configured to use a parser that does not have JSX support. I wouldn't want eslint then trying to lint those files. If somebody wants this behaviour, they can easily use `find` and `xargs`.

**Comment by nzakas** [comment]
@michaelficarra in that case, wouldn't you not have `.jsx` files then? I'm assuming that the two go together: you're using JSX so you have `jsx:true` and `.jsx` files in the project. If you aren't using JSX, then likely you don't have `.jsx` files in your project, so there's no harm.

Another option is placing this info into `.eslintrc`, but that opens up the question of what does it mean to cascade extensions?

**Comment by michaelficarra** [comment]
I was thinking about the case when a `.jsx` file was a fixture/resource, not a project source file, but that may only be common in my world working on parsers and such all day long.

**Comment by astephenb** [comment]
@nzakas I would prefer a configuration option in `.eslintrc`. As to how to handle cascading extensions, would it not make sense to overwrite existing extensions like other rules?

**Comment by nzakas** [comment]
@astephenb it's a bit complicated to do that. Right now, a list of files to check is generated upfront, so it would have to be changed to be calculated as we came across configs. This would be a significant refactor, which is why I'm hesitating. Unless someone wants to do the work to investigate the details, I'm not sure this will get done.

**Comment by iamstarkov** [comment]
I’m also want to have ability to specify file extensions in config. It is rather convenient to mirror cli options into config file. though i dont have time anto investigate.

**Comment by nzakas** [comment]
We will look into this after 2.0.0, unless someone wants to investigate in the meantime.

**Comment by kachkaev** [comment]
Any updates on this one?
Current behaviour seems a bit strange. Despite that I have React plugin installed for eslint and despite that I see the errors and warnings in .jsx files while in Atom, running `eslint ./` in the Terminal does not do the same by default. It is either necessary to type extra characters each time (`eslint --ext js,jsx ./`) or proxy linting in package.json:

```json
"scripts": {
    "lint": "eslint --ext js,jsx . || true"
  },
```
```bash
npm run-script lint
```

Both options are not ideal as they require longer commands.

How about just making .jsx a default extension?

**Comment by nzakas** [comment]
We don't plan to address this until we look at redoing how configuration is done, so no immediate changes are expected here.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
